config/config.go

```diff
@@ -493,7 +493,7 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	if config.RemoteState != nil {
 		includedConfig.RemoteState = config.RemoteState
 	}
-	if config.PreventDestroy {
+	if config.PreventDestroy != includedConfig.PreventDestroy{
```

Comment:
Ah, nice find. One NIT: 

```suggestion
	if config.PreventDestroy != includedConfig.PreventDestroy {
```